### PR TITLE
Read superblock when mounting UFS2 file system

### PR DESCRIPTION
**Pull request by gmacd**
Now focusing only on reading from within kernel.  If it ever needs to be done from user-space, we need to unpick it.

### CODE REVIEW

File: sys/src/9/ufs/ufs_harvey.h [hunk 1]
```diff
@@ -9,19 +9,52 @@
...
-	//struct mount*	v_mount;
-};
+typedef struct vnode {
+	inode	*v_data;
+	//MountPoint	*v_mount;
+} vnode;
+
+
+// Not sure we even need this - if not we can remove it later.
+typedef struct thread {
```
Comment (rminnich) [inline sys/src/9/ufs/ufs_harvey.h]: it doesn't make a lot of sense in the kernel ... in harvey, there's no such thing as a kernel level thread. Just processes.

File: sys/src/9/ufs/ffs/ffs_vfsops.c [hunk 2]
```diff
@@ -773,13 +790,16 @@ ffs_mountfs (struct vnode *devvp, struct mount *mp, struct thread *td)
...
 			break;
-		brelse(bp);
-		bp = nil;
 	}
+
 	if (sblock_try[i] == -1) {
-		error = EINVAL;		/* XXX needs translation */
+		//error = EINVAL;		/* XXX needs translation */
+		// TODO HARVEY error string?
+		error = -1;
```
Comment (rminnich) [inline sys/src/9/ufs/ffs/ffs_vfsops.c]: should be using error() here but we can always fix that up later. The goto stuff is not really how it should be done but I realize this comes from unix.

File: sys/src/9/ufs/ufs_harvey.h (hunk 1 above)
Comment (gmacd) [inline sys/src/9/ufs/ufs_harvey.h]: Yeah, I was wondering if we'd need to pass some different info down here - e.g. a process ID, so I left it in for now.

File: sys/src/9/ufs/ffs/ffs_vfsops.c (hunk 2 above)
Comment (gmacd) [inline sys/src/9/ufs/ffs/ffs_vfsops.c]: Agreed, though in these cases I'm reluctant to change too much right now.  Also, I don't want to break the flow of code for error handling, especially for deeper errors that might bubble up.

**Pull request merged**